core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -34,12 +34,15 @@ def __init__(self) -> None:
         self.subprocess: Optional[Any] = None
         self.firmware_download = FirmwareDownload()
 
-    def run(self) -> None:
+    def run(self, sitl: bool = False) -> None:
         ArduPilotManager.check_running_as_root()
 
-        while not self.start_board(BoardDetector.detect()):
-            logger.warning("Flight controller board not detected, will try again.")
-            time.sleep(2)
+        if sitl:
+            self.start_sitl()
```

Comment:
Is better to have two different runs than a run with a bool specific for sitl, this makes the logic a bit more confuse here.

Reply:
Agree